Source/themes.cpp

```diff
@@ -372,44 +322,54 @@ BOOLEAN SpecialThemeFit(int i, int t)
 			}
 		}
 		break;
-	case THEME_GOATSHRINE:
-		if (rv) {
-			rv = TFit_GoatShrine(i);
-		}
-		break;
-	case THEME_CAULDRON:
+	case THEME_MURKYFOUNTAIN:
 		if (rv) {
 			rv = TFit_Obj5(i);
 			if (rv) {
-				cauldronFlag = 0;
+				mFountainFlag = 0;

```

Comment:
```suggestion
				mFountainFlag = FALSE;
```

Same goes for the other flags.